ccsh.lua

```diff
@@ -45,11 +49,120 @@ local internalCommands = {
    exit = function() end -- Placeholder
 }
 
+
 term.write(vars.PS1)
 
 local shift, caps, control, escaped
 local args = {[0] = ""}
-local cols = {[0] = 2}
+local cols = {[0] = #vars.PS1}
+
+local function runCommand()
+   local arg = {}
+
+   for k, v in pairs(args) do
+      arg[k] = v:gsub("\x00", "")
+   end
+
+   for _, v in pairs(split(vars.PATH, ":")) do
+      local check = fs.combine(v, arg[0])
+      if v:match("^.") == "/" then
+         check = "/" .. check
+      end
+
+      if internalCommands[arg[0]] then
+         running = coroutine.create(internalCommands[arg[0]])
+
+         coroutine.resume(running, arg)
+
+         args = {[0] = ""}
+         return
+      elseif fs.exists(check) and arg[0] ~= "" then
+         local file = fs.open(check, "r")
+
+         local env = _G
+         env.args = arg
+         env.vars = vars
+
+         running = coroutine.create(load(file.readAll(), v, "t", env))
+
+         coroutine.resume(running)
+
+         args = {[0] = ""}
+         return
+      elseif fs.exists("/" .. fs.combine(shell.dir(), arg[0])) and arg[0] ~= "" then
```

Comment:
reorder `and` for performance
```suggestion
      elseif arg[0] ~= "" and fs.exists("/" .. fs.combine(shell.dir(), arg[0])) then
```